dev/NavigationView/NavigationView.cpp

```diff
@@ -379,10 +427,354 @@ void NavigationView::OnApplyTemplate()
+    }
+    return parent;
+}
+
+winrt::ItemsRepeater NavigationView::GetParentItemsRepeaterForContainer(winrt::NavigationViewItemBase nvib)
+{
+    auto child = nvib.try_as<winrt::FrameworkElement>();
+    auto parent = GetParentForFrameworkElement(child);
+
+    if (parent != nullptr)
+    {
+        if (auto parentIR = parent.try_as<winrt::ItemsRepeater>())
+        {
+            return parentIR;
+        }
+    }
+    return nullptr;
+}
+
+winrt::IndexPath NavigationView::GetIndexPathForContainer(winrt::NavigationViewItemBase nvib)
+{
+    auto path = std::vector<int>();
+
+    auto child = (nvib).try_as<winrt::FrameworkElement>();
+    auto parent = GetParentForFrameworkElement(child);
+    if (parent == nullptr)
+    {
+        return IndexPath::CreateFromIndices(path);
+    }
+
+    while (!(parent.try_as<winrt::ItemsRepeater>()) || !IsRootItemsRepeater((parent.try_as<winrt::ItemsRepeater>()).Name()))
+    {
+        if (auto parentIR = parent.try_as<winrt::ItemsRepeater>())
+        {
+            path.insert(path.begin(), parentIR.GetElementIndex(child));
+        }
+
+        child = parent;
+        auto name = parent.Name();
+        parent = GetParentForFrameworkElement(parent);
+    }
+
+    if (auto parentIR = parent.try_as<winrt::ItemsRepeater>())
+    {
+        path.insert(path.begin(), parentIR.GetElementIndex(child));
+    }
+
+    auto rootRepeaterName = (parent.try_as<winrt::ItemsRepeater>()).Name();
+
+    // If item is in one of the disconnected ItemRepeaters, account for that in IndexPath calculations
+    if (rootRepeaterName == c_overflowRepeater)
+    {
+        // Convert index of selected item in overflow to index in datasource
+        auto containerIndex = m_topNavRepeaterOverflowView.get().GetElementIndex(child);
+        auto item = m_topDataProvider.GetOverflowItems().GetAt(containerIndex);
+        auto indexAtRoot = m_topDataProvider.IndexOf(item);
+        path.erase(path.begin());
+        path.insert(path.begin(), indexAtRoot);
+    }
+    else if (rootRepeaterName == c_topNavRepeater)
+    {
+        // Convert index of selected item in overflow to index in datasource
+        auto containerIndex = m_topNavRepeater.get().GetElementIndex(child);
+        auto item = m_topDataProvider.GetPrimaryItems().GetAt(containerIndex);
+        auto indexAtRoot = m_topDataProvider.IndexOf(item);
+        path.erase(path.begin());
+        path.insert(path.begin(), indexAtRoot);
+    }
+
+    return IndexPath::CreateFromIndices(path);
+}
+
+
+void NavigationView::RepeaterElementPrepared(winrt::ItemsRepeater ir, winrt::ItemsRepeaterElementPreparedEventArgs args)
+{
+    // This validation is only relevant outside of the Windows build where WUXC and MUXC have distinct types.
+    // Certain items are disallowed in a NavigationView's items list. Check for them.
+    if (args.Element().try_as<winrt::Windows::UI::Xaml::Controls::NavigationViewItemBase>())
```

Comment:
>winrt::Windows::UI::Xaml::Controls::NavigationViewItemBase [](start = 30, length = 58)

I'm note sure why this is needed, below this we check if it is a MUX::NavViewItemBase and no opt if it isn't. Shouldn't that be sufficient?

Reply:
I think the original intention of this was to have a way of notifying developers that they are passing in the incorrect types in a clear manner